### PR TITLE
Display BoolResult from implicit/explicit conversions for more clarity

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -3217,7 +3219,7 @@ Declaration const& TypeChecker::dereference(UserDefinedTypeName const& _typeName
 bool TypeChecker::expectType(Expression const& _expression, Type const& _expectedType)
 {
 	_expression.accept(*this);
-	if (!type(_expression)->isImplicitlyConvertibleTo(_expectedType))
+	if (BoolResult result = type(_expression)->isImplicitlyConvertibleTo(_expectedType); !result)
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]:
```suggestion
	if (
		BoolResult result = type(_expression)->isImplicitlyConvertibleTo(_expectedType);
		!result
	)
```

Comment (chriseth) [reply libsolidity/analysis/TypeChecker.cpp]: I would actually find it extremely much more readable by pulling out `result` to the function scope.

**Review by chriseth — DISMISSED** [review]
Please add changelog entry.

File: libsolidity/analysis/TypeChecker.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/analysis/TypeChecker.cpp]: I can do it, the file uses both forms.

**Review by chriseth — APPROVED** [review]

**Pull request merged**